packages/file-pipeline/src/helpers/route-cache.ts

```diff
@@ -0,0 +1,111 @@
+import path from "path"
+import File from "vinyl"
+import {RouteCacheEntry, RouteCacheInterface, RouteType, RouteVerb} from "../types"
+
+export class RouteCache implements RouteCacheInterface {
+  routeCache: Record<string, RouteCacheEntry> = {}
+
+  delete(file: File) {
+    delete this.routeCache[file.originalRelative]
+  }
+
+  normalizePath(input: string) {
+    if (path.sep === path.posix.sep) return input
+    return input.split(path.sep).join(path.posix.sep)
+  }
+
+  getUrifromPath(path: string) {
+    const findStr = "/pages"
+    const findStrIdx = path.indexOf(findStr)
+    const uri = path.substring(findStrIdx + findStr.length, path.lastIndexOf("."))
+    const uriWithoutIndex = uri.replace("/index", "")
+    return uriWithoutIndex.length > 0 ? uriWithoutIndex : "/"
+  }
+
+  getVerb(type: RouteType): RouteVerb {
+    switch (type) {
+      case "api":
+        return "*"
+      case "rpc":
+        return "post"
+      default:
+        return "get"
+    }
+  }
+
+  isErrorCode(uri: string) {
+    if (uri.length === 4) {
+      // need better way to check HTTP error code
+      const regex = /^[1-5][0-9][0-9]$/
+      return regex.test(uri.substring(1))
+    }
+    return false
+  }
+
+  add(file: File, type: RouteType) {
+    const uri = this.getUrifromPath(this.normalizePath(file.path))
+    const isErrorCode = this.isErrorCode(uri)
+    if (!isErrorCode) {
+      this.routeCache[file.originalRelative] = {
+        path: file.originalRelative,
+        uri,
+        verb: this.getVerb(type),
+        type,
+      }
+    }
+  }
+
+  filterByPath(filterFn: (a: string) => boolean) {
+    let found = []
```

Comment:
You can make this `const found = []` since you are not reassigning the variable. Here and elsewhere.

Reply:
Got it!